bootloader/bootloader.c

```diff
@@ -139,11 +139,15 @@ void __attribute__((noreturn)) __stack_chk_fail(void)
 
 int main(void)
 {
-	__stack_chk_guard = random32();
 	setup();
 	memory_protect();
 	oledInit();
 
+	// to be extra careful and heed the STM32F205xx Reference manual, Section 20.3.1
+	// we don't use the first random number generated after setting the RNGEN bit in setup
+	random32();
+	__stack_chk_guard = random32(); // this supports compiler provided unpredictable stack protection checks
```

Comment:
@mruddy Can confirm

You can use my branch [`ssp`](https://github.com/saleemrashid/trezor-mcu/tree/ssp) to mess around with it, using the app in `ssp/`. It's compiled with `-O0`. If you call `hulkSmash()` to trigger a stack smash, it displays `Hulk smash!`, if you trigger a stack smash some other way, it displays `Unknown stack smash!` and if no stack smash is triggered, it displays `No stack smash...`